client/src/components/ProtectedRoute.js

```diff
@@ -1,21 +1,52 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { Route, Redirect } from "react-router-dom";
+import Api from "../util/Api";
 
-function ProtectedRoute({ component: Component, ...rest }) {
-  if (localStorage.getItem("access_token") === null) {
-    return (
-      <Route
-        {...rest}
-        render={props => (
-          <Redirect
-            to={{ pathname: "/signin", state: { from: props.location } }}
-          />
-        )}
-      />
-    );
+function ProtectedRoute({ children, ...rest }) {
+  const [is_authenticated, checkAuth] = useState(null);
+
+  useEffect(() => {
```

Comment:
If you are getting a promise/async warning on the console, try writing the function outside useEffect and call it here, or try using an ES6 immediately invoked function expression here.

Reply:
The code I have should have the same effect as what you're saying. That warning was however, present on an older version of this code.